noSelectStar/noSelectStar.go

```diff
@@ -3,11 +3,20 @@ package noSelectStar
 import (
 	"go/ast"
 	"go/token"
+	"regexp"
 	"strings"
 
 	"golang.org/x/tools/go/analysis"
 )
 
+var selectStarRegex = regexp.MustCompile(`(?i)SELECT\s+\*`)
```

Comment:
I actually realized this wouldn't catch cases like `select a.* from Users a;`

Maybe using something like below would be more accurate?

```suggestion
var selectStarRegex = regexp.MustCompile(`(?i)SELECT\s+(.+\.)?\*`)
```

Reply:
Good idea! Fixed with improved regex and added tests for table alias patterns.